rimage/depth_map.go

```diff
@@ -117,28 +116,21 @@ func (dm *DepthMap) SubImage(r image.Rectangle) *DepthMap {
 // or if it can be converted into one.
 func ConvertImageToDepthMap(img image.Image) (*DepthMap, error) {
 	switch ii := img.(type) {
-	case *LazyEncodedImage:
-		decoded, err := decodeImage(
-			context.Background(), ii.RawData(), ii.MIMEType(), ii.Bounds().Dx(), ii.Bounds().Dy(), false)
-		if err != nil {
-			return nil, err
-		}
-		return ConvertImageToDepthMap(decoded)
 	case *DepthMap:
 		return ii, nil
 	case *imageWithDepth:
 		return ii.Depth, nil
 	case *image.Gray16:
 		return gray16ToDepthMap(ii), nil
-	case *image.NRGBA64:
-		return nrgba64ToDepthMap(ii), nil
+	case *image.NRGBA:
+		return nrgbaToDepthMap(ii), nil
 	default:
 		return nil, errors.Errorf("don't know how to make DepthMap from %T", img)
 	}
 }
 
-// nrgba64ToDepthMap creates a Depthmap from an image.NRGBA64.
-func nrgba64ToDepthMap(img *image.NRGBA64) *DepthMap {
+// nrgba64ToDepthMap creates a Depthmap from an image.NRGBA.
+func nrgbaToDepthMap(img *image.NRGBA) *DepthMap {
```

Comment:
I think this function should be completely removed

Reply:
Since NRGBA encode colors in 8-bit, a 8-bit color image will not convert properly into a depth map. with only 2^8 = 256 numbers available, the maximum depth the color image could encode is 256 mm. So if a depth image was converted to a color NRGBA, probably some destructive conversion took place.

So I would say let's remove this case. We will eventually replace it in another PR with the raw depth type